botocove/cove_host_account.py

```diff
@@ -229,21 +264,37 @@ def _get_accounts_by_organization_units(
         return account_list
 
     def _get_active_org_accounts(self) -> Set[str]:
-        all_org_accounts = (
-            self.org_client.get_paginator("list_accounts")
-            .paginate()
-            .build_full_result()["Accounts"]
-        )
-        return {acc["Id"] for acc in all_org_accounts if acc["Status"] == "ACTIVE"}
+        """
+        Captures all account metadata into self.account_data for future lookup
+        and returns a set of account IDs in the AWS organization.
+        """
+        pages = self.org_client.get_paginator("list_accounts").paginate()
+        self.account_data: Dict[str, AccountTypeDef] = {
```

Comment:
In my earlier attempt I tried to do everything with pure functions. This method has the side effect of setting account_data. That's a neat trick that makes the implementation easier. I think I like it!